tests/rbd/rbd_utils.py

```diff
@@ -456,20 +464,25 @@ def disable_rbd_feature(self, pool_name, image_name, feature_name, **kwargs):
         cmd = f"rbd feature disable {pool_name}/{image_name} {feature_name}"
         return self.exec_cmd(cmd=cmd, **kwargs)
 
-    def image_resize(self, pool_name, image_name, size):
+    def image_resize(self, pool_name, image_name, size, **kw):
         """
         Fetch image info for the given pool and image
         Args:
             pool_name: name of the pool
             image_name: image of the pool
             size: new size for the image
-
+            kw: any other extra arguments including but not limited to encryption-passphrase-file
         Returns:
 
         """
-        return self.exec_cmd(
-            cmd=f"rbd resize -s {size} {pool_name}/{image_name} --allow-shrink"
-        )
+        cmd = f"rbd resize --size {size} {pool_name}/{image_name}"
+        if kw.get("passphrase"):
+            for passphrase in kw.get("passphrase"):
+                for key, value in passphrase.items():
+                    cmd += f" --{key} {value}"
```

Comment:
[Optional] can we check feasibility of having a dedicated method to iterate over passphrase and return string ?

Reply:
This might become unnecessary since I am combining device map and unmap into a single method.